### DISCUSSION TITLE
Blocking TLD on Mastodon

### DISCUSSION THREAD

**Discussion by FediVideos** [mastodon/mastodon#22750] [Q&A]
Mastodon has had a spate of nasty sites/instances all using the same free TLD recently.

Is it possible to block an entire TLD on a Mastodon instance? I've tried adding a TLD and it seemed to work, but I've got no way of testing.

**Comment by connyduck** [comment]
Yes blocking a domain also blocks all subdomains

**Comment by FediVideos** [comment]
No, I mean blocking the TLD (like .cf or whatever)? The ending part of a domain name.

Does blocking that ending block all domains using that ending?

So, for example, the .gq TLD is being almost entirely used for "shady" purposes because it is free:

https://en.wikipedia.org/wiki/.gq

> "A study conducted by internet security company Symantec found that 92% of the top 50 .gq websites were being used for "shady" purposes. Out of all of the websites on a .gq domain that were surveyed by Symantec, a whole 98.94% were being used for malicious purposes."

**Reply by YJSoft** [reply]
#12805 Looks like Domain TLD is already blockable. Just block like `tk`, `gq`.
